plugin/kubernetes/controller.go

```diff
@@ -41,9 +42,14 @@ type dnsController interface {
 	Run()
 	HasSynced() bool
 	Stop() error
+
+	// Modified returns last modified.
+	Modified() int64
 }
 
 type dnsControl struct {
+	modified int64 // modified tracks last modified
```

Comment:
please add a comment that this needs to be first because we use `sync.LoadAtomatic` and therefor it needs to be on a 8 byte boundary.